File: fsharp/AoC2021/Day08/Day08/Program.fs
open System
open System.IO

let args = Environment.GetCommandLineArgs()

if args.Length <> 2 then
    failwith "Usage: $0 <INPUT FILE>"

let input =
    File.ReadLines args.[1] |> Seq.map Solution.parse

printfn $"There are {Solution.countEasyDigits input} easy digits in the output."
printfn $"The sum of all the displays is {Solution.decodeDisplays input}."

File: csharp/AoC2021/Day08/Day08/Program.cs
// https://adventofcode.com/2021/day/8

using Day08;

if (args.Length != 1)
{
    Console.Error.WriteLine("Usage: $0 <input file>");
    Environment.Exit(1);
}

var displays = File.ReadLines(args[0]).Select(Display.Parse).ToList();

Console.WriteLine($"There are {Solution.CountEasyDigits(displays)} easy digits in the output.");
Console.WriteLine($"The sum of all the displays is {Solution.DecodeDisplays(displays)}.");
